src/aspire/basis/fle_2d_utils.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import scipy.sparse as sparse
+
+
+def transform_complex_to_real(B_conj, ells):
+    """
+    Transforms coefficients of the matrix B (see Eq. 3) from complex
+        to real. B is the linear transformation that takes FLE coefficients
+        to images.
+
+    :param B_conj: Complex conjugate of the matrix B.
+    :param ells: List of ells (Bessel function orders) in this basis.
+    :return: Transformed matrix.
+    """
+    num_basis_functions = B_conj.shape[1]
+    X = np.zeros(B_conj.shape, dtype=np.float64)
+
+    for i in range(num_basis_functions):
+        ell = ells[i]
+        if ell == 0:
+            X[:, i] = np.real(B_conj[:, i])
+        # for each ell != 0, we can populate two entries of the matrix
+        # by taking the complex conjugate of the ell with the opposite sign
+        if ell < 0:
+            s = (-1) ** np.abs(ell)
+            x0 = (B_conj[:, i] + s * B_conj[:, i + 1]) / np.sqrt(2)
+            x1 = (-B_conj[:, i] + s * B_conj[:, i + 1]) / (1j * np.sqrt(2))
+            X[:, i] = np.real(x0)
+            X[:, i + 1] = np.real(x1)
+
+    return X
+
+
+def precomp_transform_complex_to_real(ells):
+
+    count = len(ells)
+    num_nonzero = np.sum(ells == 0) + 2 * np.sum(ells != 0)
+    idx = np.zeros(num_nonzero, dtype=int)
+    jdx = np.zeros(num_nonzero, dtype=int)
+    vals = np.zeros(num_nonzero, dtype=np.complex128)
+
+    k = 0
+    for i in range(count):
+        ell = ells[i]
+        if ell == 0:
```

Comment:
ell = 0 is a special case, see Remark 1.1 of https://arxiv.org/pdf/2207.13674.pdf